front/src/pages/nft/[id].tsx

```diff
@@ -1,12 +1,36 @@
-import { useRouter } from "next/router";
 import React from "react";
+import { useRouter } from "next/router";
+import {useGetNFTById} from "../../services/nftService";
+import {GCImage} from "../../gc-components";
+import {NFTCard} from "../../schema/NFTCatalogSchema";
+import {GCGrid} from "../../gc-components/GCGrid";
+import {GCGridCol} from "../../gc-components/GCGridCol";
+import {useStyles} from "./SingleNFTPageStyles";
+
+type SingleNFTPageProps = {
+    data: NFTCard
+};
+
+const SingleNFTPage = (props: SingleNFTPageProps) => {
+    const id: number = Number(useRouter().query.id);
+    const [data, error, isLoading, isError] = useGetNFTById(id);
+    const { classes } = useStyles();
 
-type Props = {};
+    if (isLoading) {
+        return <div>Loading...</div>;
+    }
 
-const SingleNFTPage = (props: Props) => {
-    const id = useRouter().query.id;
+    if (isError) {
+        return <div>Error</div>;
+    }
 
-    return <div>NFT id: {id}</div>;
+    return <>
+        <GCGrid className={classes.grid}>
+            <GCGridCol sm={12} md={6}>
+                <GCImage src={data.nftImageUrl} alt="gun"/>
```

Comment:
Please check if it is possible to set type for data variable somehow
Now it is any

![image](https://user-images.githubusercontent.com/105725808/201876519-c71c0c47-4d98-4f2a-aa03-a8083a8b9488.png)